clientlib/src/main/proto/yelp/nrtsearch/search.proto

```diff
@@ -813,8 +816,19 @@ message Highlight {
         google.protobuf.UInt32Value max_number_of_fragments = 5;
         // Specify a query here if highlighting is desired against a different query than the search query
         Query highlight_query = 6;
-        // Set to true to highlight fields only if specified in the search query
-        bool field_match = 7;
+        // Set to true to highlight fields only if specified in the search query.
+        // This field is identical to field_match_v2 but doesn't support field override. This setting will be ignored if field_match_v2 is present.
+        bool field_match = 7 [deprecated = true];
+        // Set to true to highlight fields only if specified in the search query. (Default is false)
+        google.protobuf.BoolValue field_match_v2 = 8;
+        // Sorts highlighted fragments by score when set to true. By default, fragments will be output in the order they appear in the field. (Default is true)
+        google.protobuf.BoolValue score_ordered = 9;
+        // Select Fragmenter between span (default) and simple. This is only applicable for plain highlighters.
+        google.protobuf.StringValue fragmenter = 10;
+        // Let the fragment builder respect the multivalue fields. Each fragment won't cross multiple value fields if set true. (Default is false)
+        google.protobuf.BoolValue discrete_multivalue = 11;
+        // When highlighter_type is CUSTOM, use this string identifier to specify the highlighter. It is ignored for any other highlighter_types.
+        string custom_highlighter_name = 12;
```

Comment:
One question. As a plugin here, do we expect the user to pass custom defined params to the highlighter?

Reply:
In current design, the highlighter is a singleton instance handling all requests. We do not create highlighter per request, so we shouldn't alter it per request. Otherwise it may cause non idempotent issues.

But maybe I am not aware of some use cases where the highlighter must be altered during the execution. Please let me know.

Reply:
Not so far. Just a random question here since this is an extensible plugin. But settings should be good enough